app/utils/compare-files.ts

```diff
@@ -40,15 +50,14 @@ const sortByFile2Analyses = (
   const analysis1ComputedRisk = Number(a?.analysis2?.computedRisk);
   const analysis2ComputedRisk = Number(b?.analysis2?.computedRisk);
 
-  if (analysis2ComputedRisk < analysis1ComputedRisk) {
+  if (isNaN(analysis1ComputedRisk)) {
     return -1;
   }
 
-  if (analysis2ComputedRisk > analysis1ComputedRisk) {
-    return 1;
-  }
-
-  return 0;
+  return (
+    Number(sortPriorityMap[analysis2ComputedRisk]) -
+    Number(sortPriorityMap[analysis1ComputedRisk])
+  );
 };
```

Comment:
we can reduce this to a single function by having `analysis1` & `analysis2` parameterised